yb-voyager/cmd/conflictDetectionCache.go

```diff
@@ -226,19 +342,66 @@ func (c *ConflictDetectionCache) eventsConfict(cachedEvent *tgtdb.Event, incomin
 	}
 
 	for _, column := range uniqueKeyColumns {
-		if cachedEvent.BeforeFields[column] == nil || incomingEvent.Fields[column] == nil {
-			continue // check for the other columns(case: multiple unique keys)
+		//todo: handle nil values properly in before fields / fields
+
+		if cachedEvent.BeforeFields[column] == nil && incomingEvent.Fields[column] == nil {
+			//if c.before and i.after are nil then it can be a conflict
+			log.Infof("conflict detected for table %s, column %s, between value of event1(vsn=%d, colVal=%s) and event2(vsn=%d, colVal=%s)",
+				cachedEvent.TableNameTup.ForKey(), column, cachedEvent.Vsn, "nil", incomingEvent.Vsn, "nil")
+			return true
+		}
+
+		if cachedEvent.BeforeFields[column] == nil && incomingEvent.BeforeFields[column] != nil {
+			//if c.before and i.before are nil then it can be a conflict
+			log.Infof("conflict detected for table %s, column %s, between value of event1(vsn=%d, colVal=%s) and event2(vsn=%d, colVal=%s)",
+				cachedEvent.TableNameTup.ForKey(), column, cachedEvent.Vsn, "nil", incomingEvent.Vsn, "nil")
+			return true
 		}
 
-		if *cachedEvent.BeforeFields[column] == *incomingEvent.Fields[column] {
+		cachedEventBefore := ""
+		if cachedEvent.BeforeFields[column] != nil {
+			cachedEventBefore = *cachedEvent.BeforeFields[column]
+		}
+		incomingEventBefore := ""
+		if incomingEvent.BeforeFields[column] != nil {
+			incomingEventBefore = *incomingEvent.BeforeFields[column]
+		}
+		incomingEventAfter := ""
+		if incomingEvent.Fields[column] != nil {
+			incomingEventAfter = *incomingEvent.Fields[column]
+		}
+
+		switch true {
+		case cachedEventBefore == incomingEventAfter:
+			//If uk column is changes then it is a pure conflict
+			//Handles all cases of UPDATE-UPDATE, UPDATE-INSERT, DELETE-INSERT, DELETE-UPDATE
+
+			//If uk is not changed but the partial predicate is updated in cached and the same uk with before predicate is inserted in the incoming event then it is a partial conflict
```

Comment:
I think "partial conflict" terminology might be confusing :D It's more a potential conflict due to partial indexes.